File: framework/languages/dotnet/e2e/DiscoveryRegistryHa/Shared/Messages.cs
namespace DiscoveryRegistryHa.Shared;

public static class DiscoveryRegistryHaNames
{
    public const string Channel = "discovery.registry.ha.profile";
}

public sealed record ProfileRequest(string Value, string Marker);

public sealed record ProfileReply(string Value, string ProviderRid, string Marker);

public sealed record EvidenceWaitRequest(string Contains, int TimeoutMilliseconds = 10000);

public sealed record TopologyReadyWaitRequest(int ReadyCount, int TimeoutMilliseconds = 10000);

public sealed record MemberEndpointWaitRequest(string Endpoint, int TimeoutMilliseconds = 10000);

File: framework/languages/node/e2e/DiscoveryRegistryHa/Shared/messages.ts
export const ChannelNames = {
  profile: 'profile'
} as const;

export const PacketNames = {
  profileReq: 'ProfileReq'
} as const;

export interface ProfileReq {
  readonly value: string;
  readonly marker?: string;
}

export interface ProfileRes {
  readonly value: string;
  readonly providerRid: string;
  readonly marker?: string;
}

export interface EvidenceWaitReq {
  readonly contains: string;
  readonly timeoutMilliseconds?: number;
}
